### ISSUE TITLE
Winget automatic release for 0.35 failed

### ISSUE THREAD

**Issue by jntrnr** [nushell/nushell#3887]
### Describe the Bug

cc @TechWatching 

From the CI:

```
Manifest saved to D:\a\nushell\nushell\manifests\n\Nushell\Nushell\0.35.0

Manifest validation succeeded: True

Submitting pull request for manifest...

Octokit.ForbiddenException: Must have admin rights to Repository.
   at Octokit.Connection.HandleErrors(IResponse response)
   at Octokit.Connection.RunRequest(IRequest request, CancellationToken cancellationToken)
   at Octokit.Connection.Run[T](IRequest request, CancellationToken cancellationToken)
   at Octokit.Connection.Delete(Uri uri)
   at Microsoft.WingetCreateCore.Common.GitHub.SubmitPRAsync(String packageId, String version, Dictionary`2 contents, Boolean submitToFork)
   at Microsoft.WingetCreateCLI.Commands.BaseCommand.GitHubSubmitManifests(Manifests manifests, String token)
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.ExecuteManifestUpdate(List`1 latestManifestContent, CommandExecutedEvent commandEvent)
   at Microsoft.WingetCreateCLI.Commands.UpdateCommand.Execute()
   at Microsoft.WingetCreateCLI.Program.Main(String[] args)
{"message":"Must have admin rights to Repository.","documentation_url":"https://docs.github.com/rest/reference/repos#delete-a-repository"}
Error: Process completed with exit code 1.
```

### To Reproduce

Release on CI

### Expected behavior

The release action should post to winget repo successfully

**Comment by TechWatching** [comment]
The problem seems to come from the Personal Access Token this time, not the script in itself. From what I understand, the PAT that was generated (whose value is in the secret NUSHELL_PAT) does not have the proper permission.

Just to be sure it was not a problem from winget create I just tried on a private repo and it worked fine: the PR was correctly created as you can see [here](https://github.com/microsoft/winget-pkgs/pull/23746).

Maybe @jntrnr you can re-generate a PAT with the following permission and set the value of NUSHELL_PAT secret with it:
![image](https://user-images.githubusercontent.com/15186176/128208974-ef37cfba-afa4-4b09-9d33-4e670b7c8a1b.png)

Once this is done, if you can re-run the job of this workflow to see if that works well it would be great.

**Comment by fdncred** [comment]
I created the NUSHELL_PAT with public_repo checked. I think that's the way it says to do it in the winget documentation, iirc. We can recreate it and try again though. @jntrnr owns the repo I think so maybe JT can generate the next one.

**Comment by TechWatching** [comment]
Yes I don't know why this would need the owner of the repo to create the PAT but I have no other idea to make it work. I don't see any difference with what I have done on my [nushell fork](https://github.com/TechWatching/nushell/runs/3133552066?check_suite_focus=true) that could explain the issue.

**Comment by jntrnr** [comment]
@TechWatching - okay, it should be updated with one of my PATs now

**Comment by TechWatching** [comment]
@jntrnr Great, can you relaunch the workflow ?
I guess you have a "Re-run jobs" button on the [workflow run](https://github.com/nushell/nushell/actions/runs/1095072223) that failed. I think this should re-run using the updated PAT, well I hope it will so that we do not have to wait for next release.

**Comment by jntrnr** [comment]
Seems like it needs more in the PAT. @TechWatching was selecting `public_repo` all you needed?

**Comment by jntrnr** [comment]
Or maybe it's using the cached one? I guess that's also possible

**Comment by TechWatching** [comment]
Yes as @fdncred said it is the only permission mentionned by the documentation. And as you can see on my [fork](https://github.com/TechWatching/nushell/runs/3133552066?check_suite_focus=true) this worked well and the PAT I generated only have the public_repo permission.

![image](https://user-images.githubusercontent.com/15186176/128322401-36895ef6-d216-4bde-8b52-4ac5f090382a.png)

Maybe it used the cached one. But in that case that means we will have to wait for release 0.36.0 to check that 😕, a bit disappointing a I guess it won't happend soon.

**Comment by TechWatching** [comment]
Unless you agree to create a 0.36.0 release just to check the triggered workflow to publish to nuget has the correct PAT and that everything work. Not a great idea but as it works on my fork that contain the same exact code for the workflow than on nushell repository, I don't see what else we can do.

I guess we could also enable the logs and add more permissions that needed to the PAT until it work, but I am afraid that won't solve the issue.

**Comment by jntrnr** [comment]
Sounds like we'll just give it another go in three weeks

**Comment by TechWatching** [comment]
Okay fine. I am a bit disappointed as this time I was pretty confident that it was going to work but I guess there are always surprise with pipelines, true for Azure Pipelines but it seems also true for GitHub Actions. 

In the meantime, I will try to ask on wingetcreate's repo why a PAT from the owner of the repo is needed for wingetcreate to work and if this error has ever happen to someone. Thanks for trying though, hope next time will be the one.

**Comment by fdncred** [comment]
ya, it stinks @TechWatching . You're trying so hard and we truely appreciate it. Thanks!  btw - i am an admin on the repo so it's uber strange if @jntrnr 's works and my pat doesn't. Mine was created like this last time.
![image](https://user-images.githubusercontent.com/343840/128362867-b9a26aec-1c54-4f35-95eb-998f7147b8a5.png)

**Comment by TechWatching** [comment]
FYI I created this [issue](https://github.com/microsoft/winget-create/issues/130) on the winget-create repo.

From what I understood, winget-create update command is using the token to fork the winget-pkgs repository, push a new branch in it with the updated manifest and submit a new PR from this branch.

![image](https://user-images.githubusercontent.com/15186176/128513610-1422804d-1ed2-46ed-9555-c4a830cbff66.png)
(this is from my fork, you can see the autogenerated branches by wingetcreate).

On your forks (@fdncred  or @jntrnr ) I can't see any auto-generated branch so I guess the push of this autogenerated branch on your fork is failing. 

![image](https://user-images.githubusercontent.com/15186176/128514020-0c57b3fd-f922-4c90-a5bb-8c742d63d315.png)
 
 Hence I guess the issue has nothing to do with ownership of Nushell repository as you said Darren but more a problem to push some code to your forks. That does not really help to solve the issue, just to understand what is happening.

**Comment by fdncred** [comment]
I thought I had deleted my winget repo, I'm guessing it recreated the fork when it was using the PAT i generated. Now that it's using @jntrnr 's his should have the autogenerated one, weird that it does not.  One work around is for @TechWatching to give us his PAT that is working and we put it in the repo, although I'm not sure that's a good long term solution.

**Comment by TechWatching** [comment]
I deleted all my PAT and recreated one with only the public_repo permission and replace my secret with this new value just in case.
![image](https://user-images.githubusercontent.com/15186176/129797299-9075c20d-6f9b-4d13-a0a0-15e5a4226117.png)
Everything worked again and a PR was created when  I launched a fake release on my nushell fork.

@fdncred  Would you mind trying to execute the following commands locally with your PAT in order to check if you have the same issue with your PAT than on the github actions workflow (replace the *** of the token by your PAT value)?
```
$tagName="0.101.0"
$installerUrl="https://github.com/TechWatching/nushell/releases/download/0.101.0/nu_0_101_0_windows.msi"
$token="*******"
wingetcreate update Nushell.Nushell -s -v $tagName -u $installerUrl -t $token
```

**Comment by fdncred** [comment]
This is what I get
```
 ~  source  repos  forks  nushell  main  ﲵ pwsh  $tagName="0.101.0"
 ~  source  repos  forks  nushell  main  ﲵ pwsh  $installerUrl="https://github.com/TechWatching/nushell/releases/download/0.101.0/nu_0_101_0_windows.msi"
 ~  source  repos  forks  nushell  main  ﲵ pwsh  $token="my pat was here"
 ~  source  repos  forks  nushell  main  ﲵ pwsh  wingetcreate update Nushell.Nushell -s -v $tagName -u $installerUrl -t $token

Retrieving latest manifest for Nushell.Nushell
Downloading and parsing: https://github.com/TechWatching/nushell/releases/download/0.101.0/nu_0_101_0_windows.msi...

Generating a preview of your manifests...
Version manifest preview:
# Created using wingetcreate 0.2.0.29
# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.0.0.schema.json

PackageIdentifier: Nushell.Nushell
PackageVersion: 0.101.0
DefaultLocale: en-US
ManifestType: version
ManifestVersion: 1.0.0


Installer manifest preview:
# Created using wingetcreate 0.2.0.29
# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.0.0.schema.json

PackageIdentifier: Nushell.Nushell
PackageVersion: 0.101.0
MinimumOSVersion: 10.0.0.0
InstallModes:
- interactive
- silent
- silentWithProgress
Commands:
- nu
Installers:
- InstallerLocale: en-US
  Architecture: x64
  InstallerType: msi
  Scope: machine
  InstallerUrl: https://github.com/TechWatching/nushell/releases/download/0.101.0/nu_0_101_0_windows.msi
  InstallerSha256: 12B9256C6197FC3BE5376309295E5424C6E26D92EA2AEBC9B7D9C2E2EF8FE860
  UpgradeBehavior: install
  ProductCode: '{29F62B60-B177-40E1-B8F4-28144891E1D6}'
ManifestType: installer
ManifestVersion: 1.0.0


Default locale manifest preview:
# Created using wingetcreate 0.2.0.29
# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.0.0.schema.json

PackageIdentifier: Nushell.Nushell
PackageVersion: 0.101.0
PackageLocale: en-US
Publisher: The Nu Project Contributors
PublisherUrl: https://github.com/nushell/nushell
PublisherSupportUrl: https://github.com/nushell/nushell/issues
Author: The Nu Project Contributors
PackageName: nu
PackageUrl: https://github.com/nushell/nushell
License: MIT License
LicenseUrl: https://raw.githubusercontent.com/nushell/nushell/main/LICENSE
Copyright: Copyright (c) 2019 - 2021 Yehuda Katz, Jonathan Turner
CopyrightUrl: https://raw.githubusercontent.com/nushell/nushell/main/LICENSE
ShortDescription: A new type of shell
Moniker: nushell
Tags:
- shell
- rust
ManifestType: defaultLocale
ManifestVersion: 1.0.0



Manifest saved to C:\Users\dschroeder\source\repos\forks\nushell\manifests\n\Nushell\Nushell\0.101.0

Manifest validation succeeded: True

Submitting pull request for manifest...

Pull request can be found here: https://github.com/microsoft/winget-pkgs/pull/25404
```

it seems like it worked?

**Comment by TechWatching** [comment]
Yes it did work so I don't understand why it did not in the pipeline, the script is the same... 

Could you check just in case on nushell/nushell that there is not typo in the secret name `NUSHELL_PAT` ? I don't have many other ideas: it works locally with my token, same for you, it works in a github action workflow on my nushell fork but it does not want to work on nushell/nushell .... 

We are probably missing something obvious but I can't see what.

**Comment by fdncred** [comment]
![image](https://user-images.githubusercontent.com/343840/130070172-aca5201a-0c17-4b1a-9326-ef9a12a5293d.png)

I wonder if the pat needs to be created by user/org `nushell`, isn't that what the first `nushell` is in github/nushell/nushell?

**Comment by TechWatching** [comment]
On the [issue I created on wingetcreate repo](https://github.com/microsoft/winget-create/issues/130#issuecomment-902270670), a maintainer is suggesting to give more permissions (like the delete repo permission) to the PAT in order to see what is the real issue (which is hidden by the delete permission error). Interesting but unfortunately if we do that we only see the error in next nushell release, try something to correct it the next next release (so another 3 weeks). I don't know what you think about that.

@jntrnr would you mind doing the same [test](https://github.com/nushell/nushell/issues/3887#issuecomment-900623162) that @fdncred did locally ?

**Comment by fdncred** [comment]
give it `delete repo permission`? i'll pass on that. i'm not sure it's wise to give a bot the ability to delete a repo.

**Comment by TechWatching** [comment]
I guess we can close the issue now that the workflow worked correctly. Thanks a lot for your patience :).

FYI it was due to a [bug in wingetcreate](https://github.com/microsoft/winget-create/issues/130#issuecomment-904210305):

> I have linked the PR to address this issue. Looks like there was a timing issue when trying to immediately reference the forked repo that was just created. If a forked repo already exists, this issue doesn't repro. I believe if you were to try triggering the workflow again and the owner of the PAT already has a fork of winget-pkgs, it should work without issues.
> 
> To answer your question, yes, we would only delete the fork of the winget-pkgs repo **if the PR submission fails and we created the fork for you**. So if the fork had already existed but the PR submission failed, we still wouldn't delete the fork.
> 
> Once my PR gets checked in, we hope to get a new release out soon that can resolve your issues. Thanks for your help with this confusing issue and sorry for the inconvenience it may have caused you.

**Comment by fdncred** [comment]
Thanks for all your help @TechWatching! We sincerely appreciate it.
